dill/_dill.py

```diff
@@ -1700,6 +1700,27 @@ def _get_typedict_type(cls, clsdict, attrs, postproc_list):
         # clsdict.pop('__prepare__', None)
     return clsdict, attrs
 
+def _get_typedict_abc(obj, _dict, attrs, postproc_list):
+    if hasattr(abc, '_get_dump'):
```

Comment:
It doesn't in PyPy. PyPy's implementation of `abc` is similar to the CPython 3.6 and older implementation because it is compiled directly into a binary by the RPython compiler and the point of PyPy is to implement Python in Python, so rewriting it in C doesn't make sense for them.

Reply:
I checked `pypy` along with `python`, and found the same.  `pypy` lags changes in `python`, but it seems that for the last few versions of `pypy`, it's there.
```
Python 3.7.13 (7e0ae751533460d5f89f3ac48ce366d8642d1db5, Mar 29 2022, 06:30:54)
[PyPy 7.3.9 with GCC Apple LLVM 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>>> import abc
>>>> abc._get_dump
<built-in function _get_dump>
>>>> 
```

So, are you saying we need the code block due to older releases of `pypy`?

Reply:
Hmm. That is interesting. Yes. There are some older PyPy builds for Python 3.7 that do not have `_get_dump`. I guess they added a `_get_dump` function to make the API identical to the CPython version. Maybe we can remove this if statement?